dataproxy/service.go

```diff
@@ -231,9 +247,114 @@ func createStorageLocation(ctx context.Context, store *storage.DataStore,
 	return storagePath, nil
 }
 
+func (s Service) validateResolveArtifactRequest(req *service.GetDataRequest) error {
+	if req.GetFlyteUrl() == "" {
+		return fmt.Errorf("source is required. Provided empty string")
+	}
+	if !strings.HasPrefix(req.GetFlyteUrl(), "flyte://") {
+		return fmt.Errorf("request does not start with the correct prefix")
+	}
+
+	return nil
+}
+
+func (s Service) GetTaskExecutionID(ctx context.Context, attempt int, nodeExecID core.NodeExecutionIdentifier) (*core.TaskExecutionIdentifier, error) {
+	taskExecs, err := s.taskExecutionManager.ListTaskExecutions(ctx, admin.TaskExecutionListRequest{
+		NodeExecutionId: &nodeExecID,
+		Limit:           1,
+		Filters:         fmt.Sprintf("eq(retry_attempt,%s)", strconv.Itoa(attempt)),
```

Comment:
Is there a strongly typed way of doing this?

Reply:
not really... i think we should punt on this.